<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>a2ihelper.call_reditools &mdash; a2iHelperPy 0.0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css" />

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/sphinx_highlight.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            a2iHelperPy
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../modules.html">a2ihelper</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">a2iHelperPy</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../index.html">Module code</a></li>
      <li class="breadcrumb-item active">a2ihelper.call_reditools</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for a2ihelper.call_reditools</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">subprocess</span>
<span class="kn">from</span> <span class="nn">multiprocessing</span> <span class="kn">import</span> <span class="n">Pool</span>
<span class="kn">import</span> <span class="nn">warnings</span>
<span class="kn">import</span> <span class="nn">itertools</span>
<span class="c1"># import pandas as pd</span>

<div class="viewcode-block" id="run_per_gene_position"><a class="viewcode-back" href="../../a2ihelper.html#a2ihelper.call_reditools.run_per_gene_position">[docs]</a><span class="k">def</span> <span class="nf">run_per_gene_position</span><span class="p">(</span><span class="n">gene_position</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">in_bam_file</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">path_out_res</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">ref_genome_file</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">path_reditools</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">reditools_options</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Run reditools2.0 via Python</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    gene_position : str</span>
<span class="sd">        coordinate of chromossome:start-end positions to search editing sites.</span>
<span class="sd">        Example: &#39;chr2:122147686-122153083&#39;. One can get the coordinates from gene symbol using get_genes_positions function.</span>
<span class="sd">    in_bam_file : str</span>
<span class="sd">        full input BAM file path</span>
<span class="sd">    path_out_res: str</span>
<span class="sd">        output directory</span>
<span class="sd">    ref_genome_file: str</span>
<span class="sd">        full reference FASTA file path. Must the same used to build the aligned bam file.</span>
<span class="sd">    path_reditools: str</span>
<span class="sd">        full directory where reditools.py is installed. Usually is in similiar path &#39;/../reditools2.0/src/cineca&#39;</span>
<span class="sd">    reditools_options: str</span>
<span class="sd">        optional arguments to run reditools2. All the options are expalined in https://github.com/BioinfoUNIBA/REDItools2</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    None</span>
<span class="sd">        it doesn&#39;t return nothing, just run reditools2</span>

<span class="sd">    Example</span>
<span class="sd">    -------</span>
<span class="sd">        Using the toyfile from https://github.com/guilhermetabordaribas/a2iHelperPy</span>

<span class="sd">        &gt;&gt;&gt; gene_position = &#39;chr2:122147686-122153083&#39;</span>
<span class="sd">        &gt;&gt;&gt; in_bam_file = &#39;/.../sample1.sortedByCoord.out.bam&#39;</span>
<span class="sd">        &gt;&gt;&gt; path_out_res = &#39;/.../out/&#39;</span>
<span class="sd">        &gt;&gt;&gt; ref_genome_file = &#39;/.../GRCh38.p14.genome.fa&#39;</span>
<span class="sd">        &gt;&gt;&gt; path_reditools = &#39;/.../reditools2.0/src/cineca/&#39;</span>
<span class="sd">        &gt;&gt;&gt; reditools_options = &#39;--strict&#39;</span>
<span class="sd">        &gt;&gt;&gt; run_per_gene_position(gene_position, in_bam_file, path_out_res, ref_genome_file, path_reditools, reditools_options=&#39;--strict&#39;)</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">out_file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span> <span class="n">path_out_res</span><span class="p">,</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">in_bam_file</span><span class="p">)</span><span class="o">+</span><span class="s1">&#39;_&#39;</span><span class="o">+</span><span class="n">gene_position</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">,</span><span class="s1">&#39;_&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;-&#39;</span><span class="p">,</span><span class="s1">&#39;_&#39;</span><span class="p">)</span><span class="o">+</span><span class="s1">&#39;_RES.tsv&#39;</span> <span class="p">)</span>
    <span class="n">cmd_list</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;python&#39;</span><span class="p">,</span> <span class="s1">&#39;reditools.py&#39;</span><span class="p">,</span> <span class="s1">&#39;-f&#39;</span><span class="p">,</span> <span class="n">in_bam_file</span><span class="p">,</span> <span class="s1">&#39;-r&#39;</span><span class="p">,</span> <span class="n">ref_genome_file</span><span class="p">,</span> <span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="n">out_file</span><span class="p">,</span> <span class="s1">&#39;-g&#39;</span><span class="p">,</span> <span class="n">gene_position</span><span class="p">]</span>
    <span class="k">if</span> <span class="n">reditools_options</span><span class="p">:</span>
        <span class="n">cmd_list</span> <span class="o">+=</span> <span class="n">reditools_options</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">)</span>
    <span class="n">subprocess</span><span class="o">.</span><span class="n">call</span><span class="p">(</span><span class="n">cmd_list</span><span class="p">,</span> <span class="n">cwd</span><span class="o">=</span><span class="n">path_reditools</span><span class="p">,</span> <span class="n">stdout</span><span class="o">=</span><span class="n">subprocess</span><span class="o">.</span><span class="n">PIPE</span><span class="p">)</span></div>

<div class="viewcode-block" id="run_per_gene_position_list"><a class="viewcode-back" href="../../a2ihelper.html#a2ihelper.call_reditools.run_per_gene_position_list">[docs]</a><span class="k">def</span> <span class="nf">run_per_gene_position_list</span><span class="p">(</span><span class="n">genes_positions</span><span class="p">:</span> <span class="nb">list</span><span class="p">,</span> <span class="n">in_bam_file</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">path_out_res</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">ref_genome_file</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">path_reditools</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">reditools_options</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">n_jobs</span><span class="o">=</span><span class="mi">4</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Run run_per_gene_position for a list of gense coordinates (genes_positions)</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    gene_position: list</span>
<span class="sd">        list of coordinates of chromossome:start-end positions to search editing sites.</span>
<span class="sd">        Example: [&#39;chr2:122147686-122153083&#39;, &#39;chr18:60803848-60812646&#39;, &#39;chr6:65671590-65712326&#39;].</span>
<span class="sd">        One can get the coordinates from gene symbol using get_genes_positions function.</span>
<span class="sd">    in_bam_file: str</span>
<span class="sd">        full input file BAM path</span>
<span class="sd">    path_out_res: str</span>
<span class="sd">        output directory</span>
<span class="sd">    ref_genome_file: str</span>
<span class="sd">        full reference FASTA file path. Must the same used to build the aligned bam file.</span>
<span class="sd">    path_reditools: str</span>
<span class="sd">        full directory where reditools.py is installed. Usually is in similiar path &#39;/../reditools2.0/src/cineca&#39;</span>
<span class="sd">    reditools_options: str</span>
<span class="sd">        optional arguments to run reditools2. All the options are expalined in https://github.com/BioinfoUNIBA/REDItools2</span>
<span class="sd">    n_jobs: int</span>
<span class="sd">        number of jobs in parallel</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    None</span>
<span class="sd">        it doesn&#39;t return nothing, just run reditools2 for a list o coordinates</span>

<span class="sd">    Example</span>
<span class="sd">    -------</span>
<span class="sd">        Using the toyfile from https://github.com/guilhermetabordaribas/a2iHelperPy</span>

<span class="sd">        &gt;&gt;&gt; genes_positions = [&#39;chr2:122147686-122153083&#39;, &#39;chr6:65671590-65712326&#39;, &#39;chr15:78191114-78206400&#39;]</span>
<span class="sd">        &gt;&gt;&gt; in_bam_file = &#39;/.../sample1.sortedByCoord.out.bam&#39;</span>
<span class="sd">        &gt;&gt;&gt; path_out_res = &#39;/.../out/&#39;</span>
<span class="sd">        &gt;&gt;&gt; ref_genome_file = &#39;/.../GRCh38.p14.genome.fa&#39;</span>
<span class="sd">        &gt;&gt;&gt; path_reditools = &#39;/.../reditools2.0/src/cineca/&#39;</span>
<span class="sd">        &gt;&gt;&gt; reditools_options = &#39;--strict&#39;</span>
<span class="sd">        &gt;&gt;&gt; run_per_gene_position_list(genes_positions, in_bam_file, path_out_res, ref_genome_file, path_reditools, reditools_options=&#39;--strict&#39;, n_jobs=4)</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">arguments_list</span> <span class="o">=</span> <span class="nb">zip</span><span class="p">(</span><span class="n">genes_positions</span><span class="p">,</span> <span class="n">itertools</span><span class="o">.</span><span class="n">repeat</span><span class="p">(</span><span class="n">in_bam_file</span><span class="p">),</span> <span class="n">itertools</span><span class="o">.</span><span class="n">repeat</span><span class="p">(</span><span class="n">path_out_res</span><span class="p">),</span> <span class="n">itertools</span><span class="o">.</span><span class="n">repeat</span><span class="p">(</span><span class="n">ref_genome_file</span><span class="p">),</span> <span class="n">itertools</span><span class="o">.</span><span class="n">repeat</span><span class="p">(</span><span class="n">path_reditools</span><span class="p">),</span> <span class="n">itertools</span><span class="o">.</span><span class="n">repeat</span><span class="p">(</span><span class="n">reditools_options</span><span class="p">))</span>
    <span class="k">with</span> <span class="n">Pool</span><span class="p">(</span><span class="n">processes</span><span class="o">=</span><span class="n">n_jobs</span><span class="p">)</span> <span class="k">as</span> <span class="n">p</span><span class="p">:</span>
        <span class="n">p</span><span class="o">.</span><span class="n">starmap</span><span class="p">(</span><span class="n">run_per_gene_position</span><span class="p">,</span> <span class="n">arguments_list</span><span class="p">)</span></div>

<div class="viewcode-block" id="get_genes_positions"><a class="viewcode-back" href="../../a2ihelper.html#a2ihelper.call_reditools.get_genes_positions">[docs]</a><span class="k">def</span> <span class="nf">get_genes_positions</span><span class="p">(</span><span class="n">genes</span><span class="p">,</span> <span class="n">path_ref_annotation</span><span class="p">,</span> <span class="n">gzip_file</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">list</span><span class="p">:</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Return the coordinates of a gene symbol from a GTF file. It can be used as input to run_per_gene_position_list.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    genes: list</span>
<span class="sd">        list of genes to get coordinates</span>
<span class="sd">    path_ref_annotation: str</span>
<span class="sd">        full reference GTF file path.</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">        list</span>
<span class="sd">            a list of coordinates of gene symbol (chr:start-end).</span>

<span class="sd">    Example</span>
<span class="sd">    -------</span>
<span class="sd">        Using the GTF file from gencode</span>

<span class="sd">        &gt;&gt;&gt; get_genes_positions([&#39;B2m&#39;, &#39;Apol1&#39;], &#39;/.../GRCh38.p14.genome.fa&#39;)</span>
<span class="sd">        [&#39;chr2:122147686-122153083&#39;, &#39;chr18:60803848-60812646&#39;]</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">genes_positions_list</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">if</span> <span class="n">genes</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">gzip_file</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">genes</span><span class="p">:</span>
                <span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="n">path_ref_annotation</span><span class="p">,</span><span class="s1">&#39;r&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_gtf</span><span class="p">:</span>
                    <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">f_gtf</span><span class="p">:</span>
                        <span class="k">if</span> <span class="ow">not</span> <span class="n">line</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;#&#39;</span><span class="o">.</span><span class="n">encode</span><span class="p">()):</span>
                            <span class="n">l</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">decode</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
                            <span class="n">dict_g</span> <span class="o">=</span> <span class="p">{</span> <span class="n">i</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]:</span><span class="n">i</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="p">[</span><span class="n">j</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span> <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="n">l</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">,</span><span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;&quot;&#39;</span><span class="p">,</span><span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)]</span> <span class="p">}</span>
                            <span class="k">if</span> <span class="p">(</span><span class="n">l</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="ow">and</span> <span class="p">(</span><span class="n">dict_g</span><span class="p">[</span><span class="s1">&#39;gene_name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">g</span><span class="p">):</span>
                                <span class="n">g_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">l</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">+</span><span class="s1">&#39;:&#39;</span><span class="o">+</span><span class="n">l</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="o">+</span><span class="s1">&#39;-&#39;</span><span class="o">+</span><span class="n">l</span><span class="p">[</span><span class="mi">4</span><span class="p">])</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">genes</span><span class="p">:</span>
                <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">path_ref_annotation</span><span class="p">,</span><span class="s1">&#39;r&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_gtf</span><span class="p">:</span>
                    <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">f_gtf</span><span class="p">:</span>
                        <span class="k">if</span> <span class="ow">not</span> <span class="n">line</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;#&#39;</span><span class="p">):</span>
                            <span class="n">l</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
                            <span class="n">dict_g</span> <span class="o">=</span> <span class="p">{</span> <span class="n">i</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]:</span><span class="n">i</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="p">[</span><span class="n">j</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span> <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="n">l</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">,</span><span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;&quot;&#39;</span><span class="p">,</span><span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)]</span> <span class="p">}</span>
                            <span class="k">if</span> <span class="p">(</span><span class="n">l</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="ow">and</span> <span class="p">(</span><span class="n">dict_g</span><span class="p">[</span><span class="s1">&#39;gene_name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">g</span><span class="p">):</span>
                                <span class="n">genes_positions_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">l</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">+</span><span class="s1">&#39;:&#39;</span><span class="o">+</span><span class="n">l</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="o">+</span><span class="s1">&#39;-&#39;</span><span class="o">+</span><span class="n">l</span><span class="p">[</span><span class="mi">4</span><span class="p">])</span>

    <span class="k">if</span> <span class="ow">not</span> <span class="n">genes_positions_list</span><span class="p">:</span>
        <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s1">&#39;*Returning empty list* -&gt; Positions of genes were not found in the &#39;</span><span class="o">+</span> <span class="n">path_ref_annotation</span><span class="o">+</span><span class="s1">&#39;. Please verify genes names or gtf file.&#39;</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">genes_positions_list</span></div>


<div class="viewcode-block" id="indexing_ref"><a class="viewcode-back" href="../../a2ihelper.html#a2ihelper.call_reditools.indexing_ref">[docs]</a><span class="k">def</span> <span class="nf">indexing_ref</span><span class="p">(</span><span class="n">path_ref_genome</span><span class="p">):</span>
    <span class="k">pass</span></div>
</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Guilherme Taborda Ribas.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>