<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>a2ihelper package &mdash; a2iHelperPy 0.0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="src" href="modules.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            a2iHelperPy
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="README.html">a2ihelper</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">src</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">a2ihelper package</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#submodules">Submodules</a></li>
<li class="toctree-l3"><a class="reference internal" href="#module-a2ihelper.call_reditools2">a2ihelper.call_reditools2 module</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.call_reditools2.get_genes_positions"><code class="docutils literal notranslate"><span class="pre">get_genes_positions()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.call_reditools2.get_utr_genes_positions"><code class="docutils literal notranslate"><span class="pre">get_utr_genes_positions()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.call_reditools2.indexing_ref"><code class="docutils literal notranslate"><span class="pre">indexing_ref()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.call_reditools2.run_per_gene_position"><code class="docutils literal notranslate"><span class="pre">run_per_gene_position()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.call_reditools2.run_per_gene_position_list"><code class="docutils literal notranslate"><span class="pre">run_per_gene_position_list()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-a2ihelper.filter">a2ihelper.filter module</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.call_snp_vep"><code class="docutils literal notranslate"><span class="pre">call_snp_vep()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.filter_gtest"><code class="docutils literal notranslate"><span class="pre">filter_gtest()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.filter_positions"><code class="docutils literal notranslate"><span class="pre">filter_positions()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.filter_snps_vep"><code class="docutils literal notranslate"><span class="pre">filter_snps_vep()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.independency_gtest"><code class="docutils literal notranslate"><span class="pre">independency_gtest()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.merge_files_all_regions"><code class="docutils literal notranslate"><span class="pre">merge_files_all_regions()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.merge_files_one_region"><code class="docutils literal notranslate"><span class="pre">merge_files_one_region()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.filter.pool_positions"><code class="docutils literal notranslate"><span class="pre">pool_positions()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-a2ihelper.stats">a2ihelper.stats module</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.anova_tukey_test"><code class="docutils literal notranslate"><span class="pre">anova_tukey_test()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.chi2_test"><code class="docutils literal notranslate"><span class="pre">chi2_test()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.conditon_pearson_corr"><code class="docutils literal notranslate"><span class="pre">conditon_pearson_corr()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.entropy_calculation"><code class="docutils literal notranslate"><span class="pre">entropy_calculation()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.fisher_test"><code class="docutils literal notranslate"><span class="pre">fisher_test()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.gene_pearson_corr"><code class="docutils literal notranslate"><span class="pre">gene_pearson_corr()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.kruskal_dunn_test"><code class="docutils literal notranslate"><span class="pre">kruskal_dunn_test()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.mannwhitney_test"><code class="docutils literal notranslate"><span class="pre">mannwhitney_test()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.odds_r"><code class="docutils literal notranslate"><span class="pre">odds_r()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.stats.t_student_test"><code class="docutils literal notranslate"><span class="pre">t_student_test()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-a2ihelper.plot">a2ihelper.plot module</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.boxplot"><code class="docutils literal notranslate"><span class="pre">boxplot()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.confidence_ellipse"><code class="docutils literal notranslate"><span class="pre">confidence_ellipse()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.corr_pearson_plot"><code class="docutils literal notranslate"><span class="pre">corr_pearson_plot()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.entropy_plot"><code class="docutils literal notranslate"><span class="pre">entropy_plot()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.manhattanplot"><code class="docutils literal notranslate"><span class="pre">manhattanplot()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.pca"><code class="docutils literal notranslate"><span class="pre">pca()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.tsne"><code class="docutils literal notranslate"><span class="pre">tsne()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#a2ihelper.plot.volcanoplot"><code class="docutils literal notranslate"><span class="pre">volcanoplot()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-a2ihelper">Module contents</a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">a2iHelperPy</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">src</a></li>
      <li class="breadcrumb-item active">a2ihelper package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/a2ihelper.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="a2ihelper-package">
<h1>a2ihelper package<a class="headerlink" href="#a2ihelper-package" title="Permalink to this heading"></a></h1>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this heading"></a></h2>
</section>
<section id="module-a2ihelper.call_reditools2">
<span id="a2ihelper-call-reditools2-module"></span><h2>a2ihelper.call_reditools2 module<a class="headerlink" href="#module-a2ihelper.call_reditools2" title="Permalink to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.call_reditools2.get_genes_positions">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.call_reditools2.</span></span><span class="sig-name descname"><span class="pre">get_genes_positions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_ref_annotation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gzip_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">list</span></span></span><a class="headerlink" href="#a2ihelper.call_reditools2.get_genes_positions" title="Permalink to this definition"></a></dt>
<dd><p>Return the coordinates of a gene symbol from a GTF file. It can be used as input to run_per_gene_position_list.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>genes: list</strong></dt><dd><p>list of genes to get coordinates</p>
</dd>
<dt><strong>path_ref_annotation: str</strong></dt><dd><p>full reference GTF file path.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>dict</dt><dd><p>a dict of coordinates of each gene symbol (chr:start-end).</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.call_reditools2.get_utr_genes_positions">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.call_reditools2.</span></span><span class="sig-name descname"><span class="pre">get_utr_genes_positions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_ref_annotation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gzip_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">list</span></span></span><a class="headerlink" href="#a2ihelper.call_reditools2.get_utr_genes_positions" title="Permalink to this definition"></a></dt>
<dd><p>Return the coordinates of a gene symbol from a GTF file. It can be used as input to run_per_gene_position_list.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>genes: list</strong></dt><dd><p>list of genes to get coordinates</p>
</dd>
<dt><strong>path_ref_annotation: str</strong></dt><dd><p>full reference GTF file path.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>dict</dt><dd><p>a dict of coordinates of each gene symbol (chr:start-end).</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.call_reditools2.indexing_ref">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.call_reditools2.</span></span><span class="sig-name descname"><span class="pre">indexing_ref</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path_ref_genome</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.call_reditools2.indexing_ref" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.call_reditools2.run_per_gene_position">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.call_reditools2.</span></span><span class="sig-name descname"><span class="pre">run_per_gene_position</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gene_position</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">in_bam_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_out_res</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_reditools</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reditools_options</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#a2ihelper.call_reditools2.run_per_gene_position" title="Permalink to this definition"></a></dt>
<dd><p>Run reditools2.0 via Python</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>gene_position</strong><span class="classifier">str</span></dt><dd><p>coordinate of chromossome:start-end positions to search editing sites.
Example: ‘chr2:122147686-122153083’. One can get the coordinates from gene symbol using get_genes_positions function.</p>
</dd>
<dt><strong>in_bam_file</strong><span class="classifier">str</span></dt><dd><p>full input BAM file path</p>
</dd>
<dt><strong>path_out_res: str</strong></dt><dd><p>output directory</p>
</dd>
<dt><strong>ref_genome_file: str</strong></dt><dd><p>full reference FASTA file path. Must the same used to build the aligned bam file.</p>
</dd>
<dt><strong>path_reditools: str</strong></dt><dd><p>full directory where reditools.py is installed. Usually is in similiar path ‘/../reditools2.0/src/cineca’</p>
</dd>
<dt><strong>reditools_options: str</strong></dt><dd><p>optional arguments to run reditools2. All the options are expalined in <a class="reference external" href="https://github.com/BioinfoUNIBA/REDItools2">https://github.com/BioinfoUNIBA/REDItools2</a></p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>None</dt><dd><p>it doesn’t return nothing, just run reditools2</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.call_reditools2.run_per_gene_position_list">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.call_reditools2.</span></span><span class="sig-name descname"><span class="pre">run_per_gene_position_list</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genes_positions</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">in_bam_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_out_res</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_reditools</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reditools_options</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_jobs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">4</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#a2ihelper.call_reditools2.run_per_gene_position_list" title="Permalink to this definition"></a></dt>
<dd><p>Run run_per_gene_position for a list of gense coordinates (genes_positions)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>gene_position: list</strong></dt><dd><p>list of coordinates of chromossome:start-end positions to search editing sites.
Example: [‘chr2:122147686-122153083’, ‘chr18:60803848-60812646’, ‘chr6:65671590-65712326’].
One can get the coordinates from gene symbol using get_genes_positions function.</p>
</dd>
<dt><strong>in_bam_file: str</strong></dt><dd><p>full input file BAM path</p>
</dd>
<dt><strong>path_out_res: str</strong></dt><dd><p>output directory</p>
</dd>
<dt><strong>ref_genome_file: str</strong></dt><dd><p>full reference FASTA file path. Must the same used to build the aligned bam file.</p>
</dd>
<dt><strong>path_reditools: str</strong></dt><dd><p>full directory where reditools.py is installed. Usually is in similiar path ‘/../reditools2.0/src/cineca’</p>
</dd>
<dt><strong>reditools_options: str</strong></dt><dd><p>optional arguments to run reditools2. All the options are expalined in <a class="reference external" href="https://github.com/BioinfoUNIBA/REDItools2">https://github.com/BioinfoUNIBA/REDItools2</a></p>
</dd>
<dt><strong>n_jobs: int</strong></dt><dd><p>number of jobs in parallel</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>None</dt><dd><p>it doesn’t return nothing, just run reditools2 for a list o coordinates</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-a2ihelper.filter">
<span id="a2ihelper-filter-module"></span><h2>a2ihelper.filter module<a class="headerlink" href="#module-a2ihelper.filter" title="Permalink to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.call_snp_vep">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">call_snp_vep</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">coordinates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'homo_sapiens'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sub</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'G'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.call_snp_vep" title="Permalink to this definition"></a></dt>
<dd><p>Consult SNPs in VEP (ensembl tool).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>coordinates: list</strong></dt><dd><p>List of strings to call VEP. The string must be a list of chr_coordinate (e.g: [‘9_129401662’, ‘1_6524705’]).</p>
</dd>
<dt><strong>species: str</strong></dt><dd><p>String with the species that you are requesting. It is based on Ensembl names like homo_sapiens, mus_musculus, rattus_norvegicus, zebrafish. Other genomes can be searched here: <a class="reference external" href="https://rest.ensembl.org/documentation/info/species">https://rest.ensembl.org/documentation/info/species</a></p>
</dd>
<dt><strong>sub: str</strong></dt><dd><p>Substitution base, can be G (A&gt;G) or C (T&gt;C).</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>list</dt><dd><p>List of chr_coordinate and respective rsID</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.filter_gtest">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">filter_gtest</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtest_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bh_correction</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.filter_gtest" title="Permalink to this definition"></a></dt>
<dd><p>Filter positions limiting the quantity of samples with significant p-values for independecy G-test.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df_a: df</strong></dt><dd><p>pandas DataFrame of Adenine counts (A). Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions</p>
</dd>
<dt><strong>df_g: df</strong></dt><dd><p>pandas DataFrame of Guanine (Inosine) counts (G). Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions</p>
</dd>
<dt><strong>pvalue_filter_limit: float</strong></dt><dd><p>Minimum p-value threshold to consider the chi2_test significant.</p>
</dd>
<dt><strong>gtest_filter_limit: int</strong></dt><dd><p>Number max of conditions with significant p-value frequency in one position in each contition</p>
</dd>
<dt><strong>bh_correction: bool</strong></dt><dd><p>If True, all p-values will be corrected by false_discovery_control</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>tuple</dt><dd><p>returns inputs df_a, df_g without filtered coordinates.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.filter_positions">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">filter_positions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nan_filter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nan_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zero_filter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zero_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hundred_filter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hundred_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">per_condition</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.filter_positions" title="Permalink to this definition"></a></dt>
<dd><p>Filter positions limiting the quantity of samples with nan values, zero editing and 100% of editing across samples.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>meta: df</strong></dt><dd><p>Frequency editing DataFrame of all samples analyzed. The DataFrame must be like merge_files output. The last two columns must be region and conditions.</p>
</dd>
<dt><strong>nan_filter: bool</strong></dt><dd><p>Must be True to filter quantity of nan values across samples in each contition</p>
</dd>
<dt><strong>nan_filter_limit: int</strong></dt><dd><p>Number max of samples with nan in one position in each contition</p>
</dd>
<dt><strong>zero_filter: bool</strong></dt><dd><p>Must be True to filter quantity of zero editing values across samples in each contition</p>
</dd>
<dt><strong>zero_filter_limit: int</strong></dt><dd><p>Number max of samples with zero frequency in one position in each contition</p>
</dd>
<dt><strong>hundred_filter: bool</strong></dt><dd><p>Must be True to filter quantity of 100% of editing values across samples in each contition</p>
</dd>
<dt><strong>hundred_filter_limit: int</strong></dt><dd><p>Number max of samples with 100% frequency in one position in each contition</p>
</dd>
<dt><strong>per_condition: bool</strong></dt><dd><p>Must be True to filter quantity by condition individually. And False to filter by all samples.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>df</dt><dd><p>DataFrame without filtered positions</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.filter_snps_vep">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">filter_snps_vep</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'homo_sapiens'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sub</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'G'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.filter_snps_vep" title="Permalink to this definition"></a></dt>
<dd><p>Filter positions by consulting SNPs in VEP (ensembl tool).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df: DataFrame</strong></dt><dd><p>Frequency editing DataFrame of all samples analyzed. The DataFrame must be like merge_files output. The columns must be chr_coordinate (e.g: 9_129401662). The last two columns must be region and conditions.</p>
</dd>
<dt><strong>species: str</strong></dt><dd><p>String with the species that you are requesting. It is based on Ensembl names like homo_sapiens, mus_musculus, rattus_norvegicus, zebrafish. Other genomes can be searched here: <a class="reference external" href="https://rest.ensembl.org/documentation/info/species">https://rest.ensembl.org/documentation/info/species</a></p>
</dd>
<dt><strong>sub: str</strong></dt><dd><p>Substitution base, can be G (A&gt;G) or C (T&gt;C).</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>df_1</dt><dd><p>DataFrame df without filtered positions</p>
</dd>
<dt>df_2</dt><dd><p>DataFrame with positions and rsID</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.independency_gtest">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">independency_gtest</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_pvalue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.independency_gtest" title="Permalink to this definition"></a></dt>
<dd><p>Perform independecy G-test to verify if there’s deviation from the expected proportions and significant variation among the replicates.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df_a: df</strong></dt><dd><p>pandas DataFrame of Adenine counts (A). Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions</p>
</dd>
<dt><strong>df_g: df</strong></dt><dd><p>pandas DataFrame of Guanine (Inosine) counts (G). Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions</p>
</dd>
<dt><strong>only_pvalue: bool</strong></dt><dd><p>If True, the function will return two DataFrames only with p-values. Otherwise will return the attributes of Chi2ContingencyResult object (<a class="reference external" href="https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html">https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html</a>).</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>df</dt><dd><p>DataFrame with scipy.stats.chi2_contingency results.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.merge_files_all_regions">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">merge_files_all_regions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">meta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">coverage_q30</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.merge_files_all_regions" title="Permalink to this definition"></a></dt>
<dd><p>Merge all RES files for ALL regions (output of REDItools2) in three pandas DataFrames of frequency or count per position.
The first DataFrame is the frequency of editing (A-G or T-C). The Second DataFrame is the count of A (or T) per position. And the last one is the count of G (or C) per postion.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>meta: df</strong></dt><dd><dl class="simple">
<dt>A pandas DataFrame with metadata information for ALL regions of interest. The first four columns are mandatory</dt><dd><p>First: Full path file names of REDItools2 results tables
Second: Samples names
Third: regions (genes symbol)
Fourth: conditions</p>
</dd>
</dl>
</dd>
<dt><strong>coverage_q30: int</strong></dt><dd><p>An integer to define the minimum of reads with quality higher than in each position</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>tuple</dt><dd><p>a tuple of three pandas DataFrames (df, df_a, df_g).
df: frequency of editing
df_a: counts of A or T
df_g: counts of G or C
region_list: list of non-empty regions</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.merge_files_one_region">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">merge_files_one_region</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">meta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">coverage_q30</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.merge_files_one_region" title="Permalink to this definition"></a></dt>
<dd><p>Merge all RES files for the same and UNIQUE region (output of REDItools2) in three pandas DataFrames of frequency or count per position.
The first DataFrame is the frequency of editing (A-G or T-C). The Second DataFrame is the count of A (or T) per position. And the last one is the count of G (or C) per postion.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>meta: df</strong></dt><dd><dl class="simple">
<dt>A pandas DataFrame with metadata information for UNIQUE region. The first four columns are mandatory</dt><dd><p>First: Full path file names of REDItools2 results tables
Second: Samples names
Third: region (gene symbol)
Fourth: conditions</p>
</dd>
</dl>
</dd>
<dt><strong>coverage_q30: int</strong></dt><dd><p>An integer to define the minimum of reads with quality higher than in each position</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>tuple</dt><dd><p>a tuple of three pandas DataFrames (df, df_a, df_g).
df: frequency of editing
df_a: counts of A or T
df_g: counts of G or C</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.filter.pool_positions">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.filter.</span></span><span class="sig-name descname"><span class="pre">pool_positions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtest_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bh_correction</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.filter.pool_positions" title="Permalink to this definition"></a></dt>
<dd><p>Pool the counts per positions and per consitions of Adenine (A) and Guanine (G) Dataframes. To improve the Fisher exact test or Chi2 test between conditions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df_a: df</strong></dt><dd><p>pandas DataFrame of Adenine counts (A). Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions</p>
</dd>
<dt><strong>df_g: df</strong></dt><dd><p>pandas DataFrame of Guanine (Inosine) counts (G). Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions</p>
</dd>
<dt><strong>pvalue_filter_limit: float</strong></dt><dd><p>Minimum p-value threshold to consider the chi2_test significant.</p>
</dd>
<dt><strong>gtest_filter_limit: int</strong></dt><dd><p>Number max of conditions with significant p-value frequency in one position in each contition</p>
</dd>
<dt><strong>bh_correction: bool</strong></dt><dd><p>If True, all p-values will be corrected by false_discovery_control</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>tuple</dt><dd><p>returns inputs df_a, df_g without filtered coordinates.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-a2ihelper.stats">
<span id="a2ihelper-stats-module"></span><h2>a2ihelper.stats module<a class="headerlink" href="#module-a2ihelper.stats" title="Permalink to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.anova_tukey_test">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">anova_tukey_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_pvalue</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit_anova</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit_tukey</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.anova_tukey_test" title="Permalink to this definition"></a></dt>
<dd><p># Need to test in more than two conditions and to return a dataframe
Anova with post-hoc test for more than two conditions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df: df</strong></dt><dd><p>pandas DataFrame of editing frequency. Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>DataFrame</dt><dd><p>returns p-values for Anova post-hoc test.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.chi2_test">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">chi2_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_pvalue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.chi2_test" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.conditon_pearson_corr">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">conditon_pearson_corr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.conditon_pearson_corr" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.entropy_calculation">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">entropy_calculation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_g</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.entropy_calculation" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.fisher_test">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">fisher_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_pvalue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.fisher_test" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.gene_pearson_corr">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">gene_pearson_corr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exprs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.gene_pearson_corr" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.kruskal_dunn_test">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">kruskal_dunn_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_pvalue</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit_kruskal</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit_dunn</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_adjust</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.kruskal_dunn_test" title="Permalink to this definition"></a></dt>
<dd><p># Still NEED to test in more than two conditions and to return a dataframe
Kruskal-Wallis with post-hoc test for more than two conditions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df: df</strong></dt><dd><p>pandas DataFrame of editing frequency. Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>DataFrame</dt><dd><p>returns p-values for Anova post-hoc test.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.mannwhitney_test">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">mannwhitney_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_pvalue</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit_wilcox</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.mannwhitney_test" title="Permalink to this definition"></a></dt>
<dd><p>Perform the Mann-Whitney U rank test on two independent samples between two conditions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df: df</strong></dt><dd><p>pandas DataFrame of editing frequency. Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>DataFrame</dt><dd><p>returns p-values for Mann-Whitney U rank test.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.odds_r">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">odds_r</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_g</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.odds_r" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.stats.t_student_test">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.stats.</span></span><span class="sig-name descname"><span class="pre">t_student_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_pvalue</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_filter_limit_ttest</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_only_significant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.stats.t_student_test" title="Permalink to this definition"></a></dt>
<dd><p>Perform the t Student test on two independent samples between two conditions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df: df</strong></dt><dd><p>pandas DataFrame of editing frequency. Rows are samples and columns are coordinates. The DataFrame must be like merge_files output. The last two columns must be region and conditions.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>DataFrame</dt><dd><p>returns p-values for t Student test.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-a2ihelper.plot">
<span id="a2ihelper-plot-module"></span><h2>a2ihelper.plot module<a class="headerlink" href="#module-a2ihelper.plot" title="Permalink to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.boxplot">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">boxplot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">positions_to_plot</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">list</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log_scale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvalue_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">tuple</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.boxplot" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.confidence_ellipse">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">confidence_ellipse</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_std</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">facecolor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'none'</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.confidence_ellipse" title="Permalink to this definition"></a></dt>
<dd><p>Create a plot of the covariance confidence ellipse of <em>x</em> and <em>y</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>x, y</strong><span class="classifier">array-like, shape (n, )</span></dt><dd><p>Input data.</p>
</dd>
<dt><strong>ax</strong><span class="classifier">matplotlib.axes.Axes</span></dt><dd><p>The axes object to draw the ellipse into.</p>
</dd>
<dt><strong>n_std</strong><span class="classifier">float</span></dt><dd><p>The number of standard deviations to determine the ellipse’s radiuses.</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Forwarded to <cite>~matplotlib.patches.Ellipse</cite></p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><dl class="simple">
<dt>matplotlib.patches.Ellipse</dt><dd></dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.corr_pearson_plot">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">corr_pearson_plot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p_corr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log_scale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_value_line</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">tuple</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.corr_pearson_plot" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.entropy_plot">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">entropy_plot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">entr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_top</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log_scale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">tuple</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.entropy_plot" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.manhattanplot">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">manhattanplot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_or</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_pv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_value_line</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_order</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">tuple</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.manhattanplot" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.pca">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">pca</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">condition</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conf_ellipse</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.pca" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.tsne">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">tsne</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">condition</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conf_ellipse</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.tsne" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="a2ihelper.plot.volcanoplot">
<span class="sig-prename descclassname"><span class="pre">a2ihelper.plot.</span></span><span class="sig-name descname"><span class="pre">volcanoplot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pv_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'padj'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pv_lim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">logFC_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'log2FoldChange'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">logFC_lim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_adjusttext</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">text</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#a2ihelper.plot.volcanoplot" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

</section>
<section id="module-a2ihelper">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-a2ihelper" title="Permalink to this heading"></a></h2>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="modules.html" class="btn btn-neutral float-left" title="src" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Guilherme Taborda Ribas.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>