<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>a2ihelper &mdash; a2iHelperPy 0.0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="src" href="modules.html" />
    <link rel="prev" title="Welcome to a2iHelperPy’s documentation!" href="index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            a2iHelperPy
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">a2ihelper</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#requirements">Requirements</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installing">Installing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#quickstart">Quickstart</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">src</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">a2iHelperPy</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">a2ihelper</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/README.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="a2ihelper">
<h1>a2ihelper<a class="headerlink" href="#a2ihelper" title="Permalink to this heading"></a></h1>
<p><strong>a2ihelper</strong> is a Python library for downstream analysis of A-to-I editing. It is build under Python 3.10.13 to analyze statistics <a class="reference external" href="https://github.com/BioinfoUNIBA/REDItools2">REDItools2</a> output files.
It is possible to analyze frequency or proportional data. The principal statistics tests used in 68 papers are included in a2iHelperPy.</p>
<section id="requirements">
<h2>Requirements<a class="headerlink" href="#requirements" title="Permalink to this heading"></a></h2>
<blockquote>
<div><ul class="simple">
<li><p>numpy (version &gt;= 1.26.4)</p></li>
<li><p>pandas (version &gt;= 2.2.0)</p></li>
<li><p>scipy (version &gt;= 1.12.0)</p></li>
<li><p>scikit-posthocs (version &gt;= 0.9.0)</p></li>
<li><p><strong>Optional plot:</strong></p>
<ul>
<li><p>matplotlib (version &gt;= 3.8.3)</p></li>
<li><p>seaborn (version &gt;= 0.13.2)</p></li>
</ul>
</li>
<li><p><strong>Optionals Editing detection:</strong></p>
<ul>
<li><p>reditools (version = 2.0)</p></li>
<li><p>pysam (versoin &gt;= 0.22.0)</p></li>
<li><p>sortedcontainers (version &gt;= 2.4.0)</p></li>
<li><p>psutil (version &gt;= 5.9.8)</p></li>
<li><p>netifaces (version &gt;= 0.11.0)</p></li>
</ul>
</li>
</ul>
</div></blockquote>
</section>
<section id="installing">
<h2>Installing<a class="headerlink" href="#installing" title="Permalink to this heading"></a></h2>
<p>We recommend use pip to install the package</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip install a2ihelper
</pre></div>
</div>
<p>But you can also use the git clone and install via setup.py</p>
</section>
<section id="quickstart">
<h2>Quickstart<a class="headerlink" href="#quickstart" title="Permalink to this heading"></a></h2>
<dl class="simple">
<dt>You need:</dt><dd><ul class="simple">
<li><p>Mapped BAM files.</p></li>
<li><p>GTF annotation files (the same version used to mapping).</p></li>
<li><p>List of genes or coordinates of interest.</p></li>
</ul>
</dd>
</dl>
<p><strong>Get coordinates from list of genes:</strong></p>
<p>You need to inform a list of genes that you want to get the coordinates and the GTF annotated path file (Make sure to set <code class="docutils literal notranslate"><span class="pre">gzip_file=True</span></code> if the file is “gzipped”). It is important to use the same file used to mapping.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">genes</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Notch1&#39;</span><span class="p">,</span><span class="s1">&#39;B2m&#39;</span><span class="p">,</span><span class="s1">&#39;Hdc&#39;</span><span class="p">,</span><span class="s1">&#39;Il1b&#39;</span><span class="p">]</span>

<span class="n">path_ref_annotation</span><span class="o">=</span><span class="s1">&#39;~/../reference_files/gencode.annotation.gtf&#39;</span>

<span class="n">gene_coord</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">call_reditools2</span><span class="o">.</span><span class="n">get_genes_positions</span><span class="p">(</span><span class="n">genes</span><span class="p">,</span> <span class="n">path_ref_annotation</span><span class="p">,</span> <span class="n">gzip_file</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">gene_coord</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="s1">&#39;Notch1&#39;</span><span class="p">:</span> <span class="s1">&#39;chr2:26457903-26516663&#39;</span><span class="p">,</span> <span class="s1">&#39;B2m&#39;</span><span class="p">:</span> <span class="s1">&#39;chr2:122147686-122153083&#39;</span><span class="p">,</span>
<span class="s1">&#39;Hdc&#39;</span><span class="p">:</span> <span class="s1">&#39;chr2:126593667-126619299&#39;</span><span class="p">,</span> <span class="s1">&#39;Il1b&#39;</span><span class="p">:</span> <span class="s1">&#39;chr2:129364570-129371139&#39;</span><span class="p">}</span>
</pre></div>
</div>
<p><strong>Running REDItools2 from a2ihelper</strong></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">genes_positions</span> <span class="o">=</span> <span class="n">gene_coord</span><span class="o">.</span><span class="n">values</span><span class="p">()</span> <span class="c1"># coordinates from gene_coord dictionary</span>
<span class="n">in_bam_file_list</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;/mnt/d/rna_editing/bams/&#39;</span><span class="o">+</span><span class="n">f</span> <span class="k">for</span> <span class="n">f</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="s1">&#39;/mnt/d/rna_editing/bams/&#39;</span><span class="p">)</span> <span class="k">if</span> <span class="n">f</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;bam&#39;</span><span class="p">)]</span> <span class="c1">#list of bam files</span>
<span class="n">path_out_res</span> <span class="o">=</span> <span class="s1">&#39;/mnt/d/rna_editing/a2i_helper_output/res/&#39;</span> <span class="c1"># directory for output files</span>
<span class="n">ref_genome_file</span> <span class="o">=</span> <span class="s1">&#39;/mnt/d/reference_files/mus_musculus/GRCm39.genome.fa&#39;</span> <span class="c1"># Reference Genome</span>
<span class="n">path_reditools</span> <span class="o">=</span> <span class="s1">&#39;/mnt/d/reditools2.0/src/cineca/&#39;</span> <span class="c1"># directory where reditools.py is installed</span>
<span class="n">reditools_options</span> <span class="o">=</span> <span class="s1">&#39;--strict&#39;</span> <span class="c1"># optional arguments separeted per single space</span>

<span class="k">for</span> <span class="n">in_bam_file</span> <span class="ow">in</span> <span class="n">in_bam_file_list</span><span class="p">:</span>
   <span class="n">a2i</span><span class="o">.</span><span class="n">call_reditools2</span><span class="o">.</span><span class="n">run_per_gene_position_list</span><span class="p">(</span><span class="n">genes_positions</span><span class="p">,</span> <span class="n">in_bam_file</span><span class="p">,</span> <span class="n">path_out_res</span><span class="p">,</span>
                                                  <span class="n">ref_genome_file</span><span class="p">,</span> <span class="n">path_reditools</span><span class="p">,</span>
                                                  <span class="n">reditools_options</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="n">n_jobs</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
</pre></div>
</div>
<p>It’ll generate the RES files in the path_out_res directory.</p>
<p>Now you need to prepare your metadata file. The first column must be the path to the RES files, the second columns must be the sample name, third must be the region, fourth the condition, and the fifth the coordinates.</p>
<p><strong>Analyzing ONE region</strong></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df</span><span class="p">,</span> <span class="n">df_a</span><span class="p">,</span> <span class="n">df_g</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">merge_files_one_region</span><span class="p">(</span><span class="n">meta</span><span class="p">[</span><span class="n">meta</span><span class="o">.</span><span class="n">region</span><span class="o">==</span><span class="s1">&#39;B2m&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>The function <em>merge_files_one_region()</em> returns three DataFrames. First is the frequency of editing, second is the Adenine counts and the third the Guanine (Inosine) counts.</p>
<p><strong>Analyzing ALL region</strong></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df</span><span class="p">,</span> <span class="n">df_a</span><span class="p">,</span> <span class="n">df_g</span><span class="p">,</span> <span class="n">region_list</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">merge_files_all_regions</span><span class="p">(</span><span class="n">meta</span><span class="p">)</span>
</pre></div>
</div>
<p>The function <em>merge_files_all_regions()</em> returns three DataFrames and one list with the sequence of genes counts. First DataFrame is the frequency of editing, second is the Adenine counts and the third the Guanine (Inosine) counts.</p>
<p><strong>Statistics for FREQUENCY</strong></p>
<p><em>Mann-Whitney U test</em></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df_pv</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">mannwhitney_test</span><span class="p">(</span><span class="n">df</span><span class="p">,</span>
                                     <span class="n">only_pvalue</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                                     <span class="n">pvalue_filter_limit</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                                     <span class="n">fdr_correction</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                                     <span class="n">fdr_filter_limit</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                                     <span class="n">return_only_significant</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
<p><em>ANOVA Tukey</em></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df_pv</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">anova_tukey_test</span><span class="p">(</span><span class="n">df</span><span class="p">,</span>
                                     <span class="n">only_pvalue</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                                     <span class="n">pvalue_filter_limit_anova</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                                     <span class="n">pvalue_filter_limit_tukey</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                                     <span class="n">return_only_significant</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
<p><em>Kruskal Dunn</em></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">df_pv</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">kruskal_dunn_test</span><span class="p">(</span><span class="n">df</span><span class="p">,</span>
                                      <span class="n">only_pvalue</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                                      <span class="n">pvalue_filter_limit_kruskal</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                                      <span class="n">pvalue_filter_limit_dunn</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                                      <span class="n">return_only_significant</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Statistics for PROPORTION</strong></p>
<p><em>Pooling replicates</em></p>
<p>May you need to pool the replicates to perform a chi-square or fisher tests. To do that you can use the <code class="docutils literal notranslate"><span class="pre">pool_positions()</span></code> to sum the coordinates by independecy G-test.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">a</span><span class="p">,</span> <span class="n">g</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">pool_positions</span><span class="p">(</span><span class="n">df_a</span><span class="p">,</span>
                                  <span class="n">df_g</span><span class="p">,</span>
                                  <span class="n">pvalue_filter_limit</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                                  <span class="n">gtest_filter_limit</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                                  <span class="n">bh_correction</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
<p><em>Chi-square test</em></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">chi</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">chi2_test</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">g</span><span class="p">,</span>
                            <span class="n">only_pvalue</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                            <span class="n">pvalue_filter_limit</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
</pre></div>
</div>
<p><em>Fisher test</em></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">fis</span> <span class="o">=</span> <span class="n">a2i</span><span class="o">.</span><span class="n">editing</span><span class="o">.</span><span class="n">fisher_test</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">g</span><span class="p">,</span>
                              <span class="n">only_pvalue</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                              <span class="n">pvalue_filter_limit</span><span class="o">=</span><span class="mf">.05</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="index.html" class="btn btn-neutral float-left" title="Welcome to a2iHelperPy’s documentation!" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="modules.html" class="btn btn-neutral float-right" title="src" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Guilherme Taborda Ribas.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>